### ISSUE TITLE
Digital Ocean does not return anything on deletion

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt#33452]
### Description of Issue/Question
When deleting a digital ocean VM on 2016.3.0 it deletes the VM but I do not get a return back from salt and there is a DNS error.

### Setup
1. Create a VM in digital ocean using `salt-cloud -p`

### Steps to Reproduce Issue
1. `salt-cloud -d ch3ll-do`
2. The VM will be deleted but we do not get a return back and there is a DNS error like below:

```
[20:58:31,232][salt.loaded.int.clouds.digital_ocean     :546 ][DEBUG   ] https://api.digitalocean.com/v2/droplets/?page=1&per_page=200
[20:58:31,716][salt.loaded.int.clouds.digital_ocean     :546 ][DEBUG   ] https://api.digitalocean.com/v2/droplets/15800925/
[20:58:31,717][salt.loaded.int.clouds.digital_ocean     :769 ][DEBUG   ] Deleting DNS records for ch3ll-do.
Error: There was an error destroying machines: An error occurred while querying DigitalOcean. HTTP Code: 404  Error: '{"id":"not_found","message":"The resource you were accessing could not be found."}'
[root@li983-166 tests]# 
```

### Versions Report

```
[root@li983-166 tests]# salt --versions-report
Salt Version:
           Salt: 2016.3.0
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.7
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Jun 17 2014, 18:11:42)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5
 
System Versions:
           dist: centos 7.0.1406 Core
        machine: x86_64
        release: 4.1.5-x86_64-linode61
         system: Linux
        version: CentOS Linux 7.0.1406 Core
```

**Comment by johje349** [comment]
Getting the same error message. Worked fine in 2015.8.10.

**Comment by iggy** [comment]
As I think @Ch3LL noticed, this looks similar to #33439 maybe, although slightly different error messages. Worth keeping an eye on either way.

**Comment by ahammond** [comment]
We just upgraded to 2016.3.1 and issue still repros.

**Comment by irmatov** [comment]
I am having this issue with SaltStack 2016.3.1+ds-1:

    salt-cloud -m testcloud.yaml -d
    The following virtual machines are set to be destroyed:
      do:
        digital_ocean:
          test1
    
    Proceed? [N/y] y
    ... proceeding
    Error: There was an error destroying machines: An error occurred while querying DigitalOcean. HTTP Code: 404  Error: '{"id":"not_found","message":"The resource you were accessing could not be found."}'

VM is deleted successfully.

**Comment by johje349** [comment]
Any patch or workaround available for this problem? My issue here is that we have a reactor on destroy which does a bunch of stuff (e.g., updates /etc/hosts on all other minions) which is never triggered.

**Comment by Ch3LL** [comment]
@johje349 there is not a current workaround that I'm aware of but this is planned to be fixed for 2016.3.2 so it is on top of our list to get fixed.
